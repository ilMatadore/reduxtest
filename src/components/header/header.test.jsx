import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";

import  Header from './header.component';
import Item from '../item/item.component';
import Cart from '../cart/cart.component';

const mockStore = configureMockStore();
const store = mockStore({});

describe('Header component', () => {
    let wrapper;
    let mockuserLogOut, mockuserLogIn, mockclearItemFromCart, mockclearItem, mockaddItem, mockfetchCollectionStartAsync;

    beforeEach(()=> {
        mockuserLogOut = jest.fn();
        mockuserLogIn = jest.fn();
        mockclearItemFromCart = jest.fn(); 
        mockclearItem = jest.fn();
        mockaddItem = jest.fn();
        mockfetchCollectionStartAsync = jest.fn();

        const mockProps = {
            currentUser: {id:1, name: 'Walter'},
            userLogOut: mockuserLogOut,
            userLogIn: mockuserLogIn,
            hidden: true, 
            cartItems: [{id:1, name: 'Hat', price: 100}],
            cartTotal: 100,
            clearItemFromCart: mockclearItemFromCart,
            addItem: mockaddItem, 
            clearItem: mockclearItem,
            products: [{id:1, name: 'Hat', price: 100}],
            fetchCollectionStartAsync: mockfetchCollectionStartAsync
        }

        wrapper = shallow(
            <Provider store={store}>
                <Header {...mockProps} />
            </Provider>)
    })

    it('should render Header component', () => {
        expect(wrapper).toMatchSnapshot();
      });

 
})