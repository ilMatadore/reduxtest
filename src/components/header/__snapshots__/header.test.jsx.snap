// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render Header component 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(Header)
    addItem={[MockFunction]}
    cartItems={
      Array [
        Object {
          "id": 1,
          "name": "Hat",
          "price": 100,
        },
      ]
    }
    cartTotal={100}
    clearItem={[MockFunction]}
    clearItemFromCart={[MockFunction]}
    currentUser={
      Object {
        "id": 1,
        "name": "Walter",
      }
    }
    fetchCollectionStartAsync={[MockFunction]}
    hidden={true}
    products={
      Array [
        Object {
          "id": 1,
          "name": "Hat",
          "price": 100,
        },
      ]
    }
    userLogIn={[MockFunction]}
    userLogOut={[MockFunction]}
  />
</ContextProvider>
`;
